rita/src/rita_common/network_monitor/mod.rs

```diff
@@ -180,6 +184,77 @@ impl Default for NetworkMonitor {
     }
 }
 
+pub struct GetNetworkMonitor;
+
+impl Message for GetNetworkMonitor {
+    type Result = Result<NetworkMonitor, Error>;
+}
+
+impl Handler<GetNetworkMonitor> for NetworkMonitor {
+    type Result = Result<NetworkMonitor, Error>;
+
+    fn handle(&mut self, _msg: GetNetworkMonitor, _ctx: &mut Context<Self>) -> Self::Result {
+        Ok(self.clone())
+    }
+}
+
+#[derive(Default, Serialize, Copy, Clone)]
+pub struct LatencyStats {
+    avg: Option<f32>,
+    std_dev: Option<f32>,
+}
+
+#[derive(Default, Serialize, Copy, Clone)]
+pub struct PacketLossStats {
+    avg: Option<f32>,
+    five_min_avg: f32,
+}
+
+pub struct GetStats {
+    pub routes: Vec<BabelRoute>,
+}
+
+#[derive(Serialize, Default, Copy, Clone)]
+pub struct IfaceStats {
+    latency: LatencyStats,
+    packet_loss: PacketLossStats,
+}
+
+impl Message for GetStats {
+    type Result = Result<Stats, Error>;
+}
+
+pub type Stats = HashMap<String, IfaceStats>;
+
+impl Handler<GetStats> for NetworkMonitor {
+    type Result = Result<Stats, Error>;
+
+    fn handle(&mut self, msg: GetStats, _ctx: &mut Context<Self>) -> Self::Result {
+        let mut stats = Stats::new();
+
+        for route in msg.routes.iter() {
+            let latency_stats = self.latency_history.get_mut(&route.iface).unwrap();
```

Comment:
you shouldn't need to get these as mutable, you're not trying to change the history? You probably added mut here to avoid a .clone() but in this case you need one.